sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -125,8 +129,14 @@ object JdbcUtils extends Logging {
       dialect: JdbcDialect): Iterator[Byte] = {
     val conn = getConnection()
     var committed = false
+    val supportsTransactions = Try(
```

Comment:
PS I think you were commenting on the style, but I had suggested that this needed more changes since it just swallows the exception. WDYT? see https://github.com/apache/spark/pull/9973#discussion_r45894114